### ISSUE TITLE
D2Game.0x6FCF3000

### ISSUE THREAD

**Issue by silvermane-HU** [ThePhrozenKeep/D2MOO#142]
few edits that I think are correct looking at the ASM

```//D2Game.0x6FCF3000
int32_t __fastcall AIUTIL_CanUnitSwitchAi(D2UnitStrc* pUnit, int32_t nSpecialState)
{
	if (!pUnit || pUnit->dwUnitType != UNIT_MONSTER || nSpecialState > 19 || STATES_CheckState(pUnit, STATE_UNINTERRUPTABLE) || !UNITS_CanSwitchAI(pUnit->dwClassId))
	{
		return 0;
	}

	if (!(pUnit->dwFlags & UNITFLAG_CANBEATTACKED) && !SUNIT_IsDead(pUnit))
	{
		return 0;
	}

	if (MONSTERUNIQUE_CheckMonTypeFlag(pUnit, MONTYPEFLAG_UNIQUE | MONTYPEFLAG_SUPERUNIQUE))
	{
		return 0;
	}

	if (nSpecialState == 19)
	{
		return 1;
	}

	return AITHINK_CanUnitSwitchAi(pUnit, MONSTERMODE_GetMonStatsTxtRecord(pUnit->dwClassId), nSpecialState, 1);
}```

**Comment by Lectem** [comment]
It seems to remove the check on `pUnit->dwClassId` being valid, which we should keep according tot he ASM. I'll refactor the function a bit though.

**Comment by Lectem** [comment]
Fixed by d013191fda22e4cdbe17c5261b7bbd9a37804c82
